File: snobot_sim_java/java/snobot_sim_java/com/snobot/simulator/module_wrapper/RelayWrapper.java
package com.snobot.simulator.module_wrapper;

public class RelayWrapper extends ASensorWrapper
{

    private boolean mForwards;
    private boolean mReverse;

    public RelayWrapper(int aPort)
    {
        super("Relay " + aPort);
    }

    public void setRelayForwards(boolean aForwards)
    {
        mForwards = aForwards;
    }

    public void setRelayReverse(boolean aReverse)
    {
        mReverse = aReverse;
    }

    public boolean getRelayForwards()
    {
        return mForwards;
    }

    public boolean getRelayReverse()
    {
        return mReverse;
    }

}

File: snobot_sim/src/main/native/cpp/SnobotSim/ModuleWrapper/RelayWrapper.cpp
/*
 * RelayWrapper.cpp
 *
 *  Created on: May 3, 2017
 *      Author: PJ
 */

#include "SnobotSim/ModuleWrapper/RelayWrapper.h"

#include "SnobotSim/PortUnwrapper.h"

RelayWrapper::RelayWrapper(int aPort) :
        AModuleWrapper("Relay " + std::to_string(UnwrapPort(aPort))),
        mForwards(false),
        mReverse(false)
{
}

RelayWrapper::~RelayWrapper()
{
}

void RelayWrapper::SetRelayForwards(bool aOn)
{
    mForwards = aOn;
}

void RelayWrapper::SetRelayReverse(bool aOn)
{
    mReverse = aOn;
}

bool RelayWrapper::GetRelayForwards()
{
    return mForwards;
}

bool RelayWrapper::GetRelayReverse()
{
    return mReverse;
}
